Missing Status Code for HiGHS Solver Causes KeyError

<!--
Thanks for coming here to report a bug. :)

Please describe it in the sections below, fill out the correct check boxes with an "x", replacing the space inside [ ], then click the "Submit new issue" button at the bottom
-->

Details for the issue
--------------------

#### What did you do?
HiGHS solver offers an option, `mip_max_improving_cols`, to terminate early based on the number of improving solutions observed thus far. I was able to use the option via pulp's `solverParams` argument:
`model.solve(solver=pulp.HiGHS(mip_max_improving_sols = 1))`




#### What did you expect to see?
I expected to see a feasible solution after the solver has executed.

#### What did you see instead?
Although the output available showed that the problem was feasible, after solving, I saw a key error as `pulp` did not recognize the status code produced by the `highspy` library:
```python
        if obj_value == float(inf) and status in (
            HighsModelStatus.kTimeLimit,
            HighsModelStatus.kIterationLimit,
        ):
            return (
                constants.LpStatusNotSolved,
                constants.LpSolutionNoSolutionFound,
            )
        else:
>           return status_dict[status]
                   ^^^^^^^^^^^^^^^^^^^
E           KeyError: <HighsModelStatus.kSolutionLimit: 16>

.venv/lib/python3.13/site-packages/pulp/apis/highs_api.py:517: KeyError
```

I suspect this is because the latest version of PuLP [does not ](https://github.com/coin-or/pulp/blob/115f0839bf07f01272e9c05a8a09ae33b58c7cac/pulp/apis/highs_api.py#L415) include the highspy [key](https://github.com/ERGO-Code/HiGHS/blob/3c26f3306c53c31a16b54a32610ac551fe31cd40/docs/src/structures/enums.md?plain=1#L82). It looks as though PuLP covers many of the other status codes. I'm currently using highspy 1.12.

Thank you for your work on this project. Please let me know if additional information is needed.


Useful extra information
-------------------------

The info below often helps, please fill it out if you're able to. :)

#### What operating system are you using?

- [ ] Windows: ( _version:_ ___ )
- [ ] Linux: ( _distro:_ ___ )
- [x] Mac OS: ( _version:_ ___ )
- [ ] Other: ___

#### I'm using python version:

- [ ] 3.7
- [ ] 3.8
- [ ] 3.9
- [ ] 3.10
- [ ] 3.11
- [x ] Other: 3.13.5

#### I installed PuLP via:

- [ ] pypi (python -m pip install pulp)
- [ ] github (python -m pip install -U git+https://github.com/coin-or/pulp)
- [ x] Other: uv

#### Did you also

- [x ] Tried out the latest github version: https://github.com/coin-or/pulp
- [x] Searched for an existing similar issue: https://github.com/coin-or/pulp/issues?utf8=%E2%9C%93&q=is%3Aissue%20

--- a/pulp/apis/highs_api.py
+++ b/pulp/apis/highs_api.py
@@ -483,6 +483,14 @@ class HiGHS(LpSolver):
                     constants.LpStatusOptimal,
                     constants.LpSolutionIntegerFeasible,
                 ),
+                HighsModelStatus.kSolutionLimit: (
+                    constants.LpStatusOptimal,
+                    constants.LpSolutionIntegerFeasible,
+                ),
+                HighsModelStatus.kMemoryLimit: (
+                    constants.LpStatusNotSolved,
+                    constants.LpSolutionNoSolutionFound,
+                ),
                 HighsModelStatus.kUnknown: (
                     constants.LpStatusNotSolved,
                     constants.LpSolutionNoSolutionFound,
@@ -513,6 +521,7 @@ class HiGHS(LpSolver):
             if obj_value == float(inf) and status in (
                 HighsModelStatus.kTimeLimit,
                 HighsModelStatus.kIterationLimit,
+                HighsModelStatus.kSolutionLimit,
             ):
                 return (
                     constants.LpStatusNotSolved,

